### PR TITLE
supporting onyx feature in salt

### PR DESCRIPTION
**Pull request by anasbadaha**
I added support for managing onyx switches using salt

### CODE REVIEW

**Review by cachedout — APPROVED** [review]
LGTM

**Comment by rallytime** [comment]
@anasbadaha There are some lint failures: https://jenkinsci.saltstack.com/job/pr-lint/job/PR-49690/1/warnings52Result/

Can you fix them?

File: salt/grains/onyx.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+'''
+Grains for Onyx OS Switches Proxy minions
+
+.. versionadded: 2018.3.2 (Oxygen)
```
Comment (rallytime) [inline salt/grains/onyx.py]: This should actually be `Neon`. That is the current code name for the develop branch.

File: salt/proxy/onyx.py [hunk 2]
```diff
@@ -0,0 +1,439 @@
+# -*- coding: utf-8 -*-
+'''
+Proxy Minion for Onyx OS Switches
+
+.. versionadded: 2018.3.2 (Oxygen)
```
Comment (rallytime) [inline salt/proxy/onyx.py]: Here too. :)

File: salt/modules/onyx.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+'''
+Execution module for Onyx OS Switches Proxy minions
+
+.. versionadded:: 2018.3.2 (Oxygen)
```
Comment (rallytime) [inline salt/modules/onyx.py]: Same thing here.

File: salt/states/onyx.py [hunk 4]
```diff
@@ -0,0 +1,388 @@
+# -*- coding: utf-8 -*-
+'''
+State module for Onyx OS Switches Proxy minions
+
+.. versionadded: 2018.3.2 (Oxygen)
```
Comment (rallytime) [inline salt/states/onyx.py]: And this should be `Neon` as well. :]

**Review by rallytime — COMMENTED** [review]
Hi @anasbadaha - in addition to getting those lint errors cleaned up, I have some minor doc changes as well.

Thank you for contributing this! Looking forward to your updates.

**Comment by rallytime** [comment]
Hi @anasbadaha - Were you able to come back to this?

**Comment by anasbadaha** [comment]
Hi @rallytime  I have some failures here and I don't know how to resolve them because i'm new in github if you can help me i will appreciate it

**Comment by rallytime** [comment]
@anasbadaha Sure thing. :)

So the fixes that need to be made are for the pylint run, which are here: https://jenkinsci.saltstack.com/job/pr-lint/job/PR-49690/1/warnings52Result/

I usually look at the `Details` tab, which will tell you what is wrong and where.

Once you've made those fixes and commited your changes, all you have to do is push up your new commits to your feature branch, and GitHub will start the tests again automatically for you.

Please let me know if you have any other questions! :)

**Comment by anasbadaha** [comment]
Thanks a lot @rallytime will fix it and push again :)

**Comment by anasbadaha** [comment]
Hi @rallytime I saw that I have some errors on functions that I cloned them from github and didn't make any changes on them Ex: "Function user_present has 6 arguments (exceeds 4 allowed). Consider refactoring."! what do you think about this ?

**Comment by anasbadaha** [comment]
@rallytime I cloned the latest code from github from devel brach

**Comment by anasbadaha** [comment]
Will Fix Failed Checks and push again!

**Pull request closed without merging**